### DISCUSSION TITLE
Programmatically select Autocomplete options

### DISCUSSION THREAD

**Discussion by markarnolditpro** [microsoft/fluentui-blazor#1728] [Q&A]
What is the correct mechanism to pre-select options in Autocomplete from code? 

I currently just provide a list of objects to `SelectedOptions`.  Works as far as the Autocomplete UI is concerned but I'm seeing flakiness in select/unselect behavior.  Wondering if I should instead call some Autocomplete  method to select options as appropriate, but I'm not seeing a `select` method.  Perhaps I'm going about this wrong...

**Comment by dvoituron** [comment]
Yes, you can use the `SelectedOptions` attribute.

Example, adding this code in the existing default demo: https://fluentui-blazor.net/Autocomplete

```csharp
protected async override Task OnInitializedAsync()
{
    var allCountries = await Data.GetCountriesAsync();

    SelectedItems = new[]
    {
        allCountries.First(i => i.Code == "be"),
        allCountries.First(i => i.Code == "fr"),
    };
}
```

The result is...
![peek](https://github.com/microsoft/fluentui-blazor/assets/8350694/9f5458ee-c625-4bfd-a56e-e8ba5c49da44)

What is your question?

**Comment by markarnolditpro** [comment]
@dvoituron Thank you!  That was the clue I needed.  Entirely my mistake.  

I was persisting selected objects to localstorage, then deserializing and adding directly to `SelectedOptions` on next `OnInitialized`.  The problem is these are now completely separate objects, not objects in the full list of objects.
